File: _2717_semi_ordered_permutation/main.py
class Solution:
    def semiOrderedPermutation(self, nums: List[int]) -> int:
        n = len(nums)
        result = 0

        first = nums.index(1)
        last = nums.index(n)

        if first > last:
            result -= 1

        result += first + (n - 1) - last

        return result

File: _2717_semi_ordered_permutation/main.cpp
class Solution {
public:
    int semiOrderedPermutation(vector<int>& nums) {
        int first = -1;
        int last = -1;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 1) {
                first = i;
            } else if (nums[i] == nums.size()) {
                last = i;
            }
        }

        if (first < last) {
            return first + nums.size() - last - 1;
        } else {
            return first + nums.size() - last - 2;
        }
    }
};